gliderpy/__init__.py

```diff
@@ -15,3 +18,36 @@
     "plot_transect",
     "plot_ts",
 ]
+
+
+def _num_profiles(df: pd.DataFrame) -> int:
+    """Compute the number of unique glider pofiles."""
+    return len(df[["latitude", "longitude"]].value_counts())
+
+
+def _days(df: pd.DataFrame) -> pd.Timedelta:
+    """Compute the glider days."""
+    return df.index.dropna()[-1].ceil("D") - df.index.dropna()[0].floor("D")
```

Comment:
@MathewBiddle I decided to drop na here, to avoid the `Nemesis-20170512T0000` bad datetime data affecting the glider days computation, instead of dropping all rows with NaNs. That way we ensure we are computing what we can for the glider days without losing n_profiles info.

Reply:
In my process I did:

```python
df = df.dropna(subset=["time (utc)","latitude","longitude"], how="any")
```

With the assumption that if there is no time, latitude, or longitude, I didn't care about the point. After all, how useful is an observation if you don't know when or where it was collected?

Reply:
You are using any, that means if time is missing, lon, lat will be dropped. I believe that is what is happening with that glider, only some datetimes are missing.

Reply:
It is not a big deal as we will have a bias one way or the other. We just need to choose if we want to say that metrics are only valid when all data is present, or glider days may be biased due to missing data but n_profiles is not.

![Screenshot from 2025-05-23 09-14-16](https://github.com/user-attachments/assets/200e4d49-9f53-4500-a844-474454ce00a6)

Reply:
I lean towards conservative math. I prefer only valid when all data are present.

Reply:
While I agree. I'm afraid of data providers coming after me with pitchforks for throwing away their data ;-p

Jokes aside, yeah, for high level metrics, using all valid is better. We only need to document it in case one day this is a significant bias, which is not, only that glider has this peculiarity.